### ISSUE TITLE
Dataset memory bottleneck not showing in debug mode

### ISSUE THREAD

**Issue by andrebeu** [tensorflow/tensorflow#14196]
im fitting word2vec models using distributed gpus which require me to assemble multiple towers and so copy my model several times over. to load data i am using the get_next method of a contrib.dataset iterator initialized with one_shot. before starting up i run into the 2gb protobuf memory bottleneck. so in debug mode i compared 3 models: one where dataset is given full data, another where dataset is given 30% of my data and a last where dataset is given 1% of my data. the first doesnt run. but most importantly, for the second two, i compared the size of all of my tensors and they are all the same. this makes debugging difficult. any thoughts? i suspect there's something going on with folding of constants? 

Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug or a feature request.
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **TensorFlow installed from (source or binary)**:
- **TensorFlow version (use command below)**:
- **Python version**: 
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by bignamehyp** [comment]
Please provide details about what platform you are using  (operating system, architecture). Also include your TensorFlow version. Also, did you compile from source or install a binary?  Make sure you also include the exact command if possible to produce  the output included in your test case. If you are unclear what to include  see the issue template displayed in  [the Github new issue template](https://github.com/tensorflow/tensorflow/issues/new).

 We ask for this in the issue submission template, because    it is really difficult to help without that information. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue? Please update the label and/or status accordingly.

**Comment by andrebeu** [comment]
hi, 

In sum: It would be nice if the tf debugging mode _did not fold constants_. 

I was running into the 2GB protobuff limit because I was trying to load my data onto the graph. It took me the longest while to figure out what was causing the memory overload because the tensor nodes which held my data were not showing up in debugging mode. I solved it by using initializable iterator and the feed_dict mechanism. 

feel free to close this thread. 

Thanks!

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
The original poster has replied to this issue after the stat:awaiting response label was applied.

**Comment by drpngx** [comment]
Thank you @andrebeu for reporting back. Glad you solved the issue!
